go/cmd/vtctldclient/command/vreplication/lookupvindex/lookupvindex.go

```diff
@@ -0,0 +1,283 @@
+		TargetKeyspace string
+	}{}
+
+	// base is the base command for all actions related to Lookup Vindexes.
+	base = &cobra.Command{
+		Use:                   "LookupVindex --keyspace <keyspace> [command] [<vindex spec>]",
+		Short:                 "Perform commands related to creating, backfilling, and externalizing Lookup Vindexes using VReplication workflows.",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"lookupvindex"},
+		Args:                  cobra.NoArgs,
+	}
+
+	createOptions = struct {
+		Cells                        []string
+		TabletTypes                  []topodatapb.TabletType
+		TabletTypesInPreferenceOrder bool
+		ContinueAfterCopyWithOwner   bool
+	}{}
+
+	parseAndValidateInput = func(cmd *cobra.Command, args []string) error {
+		// Validate provided JSON spec.
+		jsonSpec := []byte(args[0])
+		baseOptions.Vindex = &vschemapb.Keyspace{}
+		if err := protojson.Unmarshal(jsonSpec, baseOptions.Vindex); err != nil {
+			return fmt.Errorf("invalid vindex specs: %v", err)
+		}
+		if baseOptions.Vindex == nil || len(baseOptions.Vindex.Vindexes) != 1 {
+			return fmt.Errorf("vindex spec must contain exactly one vindex")
+		}
+		vindexName := maps.Keys(baseOptions.Vindex.Vindexes)[0]
+		vindex := maps.Values(baseOptions.Vindex.Vindexes)[0]
+		var tableName string
+		var err error
+		baseOptions.TargetKeyspace, tableName, err = sqlparser.ParseTable(vindex.Params["table"])
+		if err != nil || (baseOptions.TargetKeyspace == "" || tableName == "") {
+			return fmt.Errorf("invalid vindex table name: %s, it must be in the form <keyspace>.<table>", vindex.Params["table"])
+		}
+		// Workflow name is the name of the vindex target table with a
+		// static suffix of `_vdx` added.
+		baseOptions.Workflow = fmt.Sprintf("%s_vdx", vindexName)
+		if cmd.Flags() == nil { // No specific command flags to verify.
+			return nil
+		}
+		// Only the create command has these additional flags so we
+		// only validate them if they exist for the command and they
+		// were changed from their default.
+		ttFlag := cmd.Flags().Lookup("tablet-types")
+		if ttFlag != nil && ttFlag.Changed {
+			createOptions.TabletTypes = tabletTypesDefault
+		}
+		cFlag := cmd.Flags().Lookup("cells")
+		if cFlag != nil && cFlag.Changed {
+			for i, cell := range createOptions.Cells {
+				createOptions.Cells[i] = strings.TrimSpace(cell)
+			}
+		}
+		return nil
+	}
+
+	// cancel makes a WorkflowDelete call to a vtctld.
+	cancel = &cobra.Command{
+		Use:                   "cancel",
+		Short:                 "Canel the VReplication workflow that backfills the lookup vindex.",
+		Example:               `vtctldclient --server localhost:15999 LookupVindex --keyspace customer cancel '{"sharded":true,"vindexes":{"corder_lookup":{"type":"consistent_lookup_unique","params":{"table":"customer.corder_lookup","from":"sku","to":"keyspace_id"},"owner":"corder"}},"tables":{"corder":{"column_vindexes":[{"column":"sku","name":"corder_lookup"}]}}}'`,
```

Comment:
Do we need to specify the spec for the `cancel` command? Should we just require the name of the workflow that is generated and output when the `LookupVindex` workflow is created?

Reply:
Same for `show` and `externalize`

Reply:
Yes, because the spec defines the target keyspace.

Reply:
We could allow the user to specify the values we showed them, but IMO it's easier to just modify the create command used to replace create with cancel.

Reply:
Why not similar to materialize, allow the spec to be defined separately to the workflow name:
`LookupVindex --keyspace customer --workflow corder_vdx Create [--spec=<inline-spec>] [--spec-file=<corder_vdx.json]`
`LookupVindex --keyspace customer --workflow corder_vdx Complete`
`LookupVindex --keyspace customer --workflow corder_vdx Cancel`

I know we currently add the "_vdx" by convention, but since we know have the `workflow_subtype` we don't need this anymore. Users can still use this, when they specify the workflow name.

Users can always recover the name, if they forget, by doing a `Workflow customer List` just like for any other workflow.

Reply:
`--keyspace` is where the LookupVindex is created. It may or may not be where the lookup table is created, which is where the workflow exists.

Reply:
Updated. Thanks!